File: algorithm/杭电ACM/ambition0109/1013.cpp
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2010-04-10 12:47:10
////Compiler: Visual C
//////////////////////////////////////////////////////
////Problem ID: 1013
////Problem Title: 
////Run result: Accept
////Run time:0MS
////Run memory:204KB
//////////////////System Comment End//////////////////
#include<stdio.h>
main()
{
	int i,m;
	char num[1001];
	while(scanf("%s",num),num[0]!='0'||num[1]!='\0')
	{
		for(i=0;num[i+1]!='\0';i++)
		{
			m=num[i]+num[i+1]-'0'*2;
			m=m/10+m%10;
			num[i+1]=m+'0';
		}
		printf("%c\n",num[i]);
	}
}

File: algorithm/杭电ACM/ambition0109/1013.java
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2011-04-12 19:49:44
////Compiler: Java
//////////////////////////////////////////////////////
////Problem ID: 1013
////Problem Title: 
////Run result: Accept
////Run time:78MS
////Run memory:6040KB
//////////////////System Comment End//////////////////
import java.util.Scanner;

public class Main {
	public static void main(String args[]){
		Scanner in=new Scanner(System.in);
		while(true){
			String str=in.next();
			if(str.equals("0")) break;
			int ans=0;
			for(int i=0;i<str.length();i++){
				int tmp=ans+str.charAt(i)-'0';
				tmp=tmp/10+tmp%10;
				ans=tmp;
			}
			System.out.println(ans);
		}
	}
}
